File: Hashing/characterhash.py
val = str(input("Enter the string"))

hashArr = [0] * 27


for i in val:
    hashArr[ord(i)-ord('a')] += 1 # ord() returns the ASCII value

# q = list(input("Enter the queries"))
q = int(input("Enter the query length"))
res = [0]*q
for j in range(q):
    n = str(input())
    print(n, ord(n))
    res[j] = (hashArr[ord(n)-ord('a')])
    # print(hashArr[ord(n)-ord('a')])

print(res)

print(hashArr)

File: Hashing/characterhash.c
#include <stdio.h>

int main()
{
    int n, q;
    int hash[27] = {0};
    scanf("%d", &n);
    char val[n];
    scanf("%s", &val);
    for (int i = 0; i < n; i++)
    {
        hash[val[i] - 'a'] += 1;
    }
    printf("Enter query length\n");
    scanf("%d", &q);
    int res[q];
    for (int j = 0; j < q; j++)
    {
        char inp;
        scanf(" %c", &inp);
        res[j] = hash[inp - 'a'];
    }

    for (int k = 0; k < sizeof(res) / sizeof(res[0]); k++)
    {
        printf("%d ", res[k]);
    }
    return 0;
}